doc/task-hierarchy.md

```diff
@@ -65,6 +76,7 @@
   - `Reasoning -> Reasoning with Symbols`: Tasks where symbols represent various things e.g. if X is the number of apples in the freeze today morning and Y is the number remaining after I ate a few apples, X-Y is the number of apples I ate.
   - `Reasoning -> Spatial Reasoning`
   - `Reasoning -> Temporal Reasoning`
+  - `Reasoning -> Scientific Reasoning`
```

Comment:
Not sure what this means. Could you define it?

Reply:
The type of reasoning required to be able to answer questions related to science

Reply:
So in order to answer a question like "Question: A student riding a bicycle observes that it moves faster on a smooth road than on a rough road. This happens because the smooth road has (A) less gravity(B) more gravity(C) less friction(D) more friction?", what do you think would be the type of reasoning, if not scientific?

Reply:
Ok, talked with @swarooprm and decided that this would be Reasoning -> Qualitative Reasoning. Science would be a domain like you said (it's already included). I've updated everything and this PR should be good to go. @danyaljj what do you think?